File: fixtures/projects/e2e-csharp/Assets/Scripts/Middleware/AuthMiddleware.cs
using MyApp.Constants;
using MyApp.Helpers;
using MyApp.Types;

namespace MyApp.Middleware
{
    public class AuthMiddleware
    {
        public ApiResponse Authorize(string token)
        {
            var normalized = StringHelper.NormalizeSlug(token);
            if (normalized.Length < HttpConstants.BearerTokenMinLength)
            {
                return new ApiResponse(false, "unauthorized", HttpConstants.UnauthorizedStatusCode);
            }

            return new ApiResponse(true, "authorized", HttpConstants.OkStatusCode);
        }

        public string BuildHeader(string token)
        {
            return $"Bearer {token}";
        }
    }
}

File: fixtures/projects/e2e-ruby/lib/middleware/auth_middleware.rb
require_relative "../constants/http_constants.rb"
require_relative "../helpers/string_helper.rb"

module Middleware
  class AuthMiddleware
    def initialize(app)
      @app = app
    end

    def call(env)
      token = env[HttpConstants::AUTHORIZATION_HEADER]
      return unauthorized unless token.to_s.start_with?("Bearer ")

      env["normalized_user"] = StringHelper.slugify(env["REMOTE_USER"] || "guest")
      @app.call(env)
    end

    private

    def unauthorized
      [HttpConstants::UNAUTHORIZED, {}, ["unauthorized"]]
    end
  end
end
